packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -415,6 +373,60 @@ class RenderUiKitView extends RenderBox {
   @override
   void detach() {
     GestureBinding.instance.pointerRouter.removeGlobalRoute(_handleGlobalPointerEvent);
+    super.detach();
+  }
+}
+
+/// A render object for an iOS UIKit UIView.
+///
+/// [RenderUiKitView] is responsible for sizing and displaying an iOS
+/// [UIView](https://developer.apple.com/documentation/uikit/uiview).
+///
+/// UIViews are added as sub views of the FlutterView and are composited by Quartz.
+///
+/// {@macro flutter.rendering.RenderAndroidView.layout}
+///
+/// {@macro flutter.rendering.RenderAndroidView.gestures}
+///
+/// See also:
+///
+///  * [UiKitView] which is a widget that is used to show a UIView.
+///  * [PlatformViewsService] which is a service for controlling platform views.
+class RenderUiKitView extends RenderDarwinPlatformView<UiKitViewController> {
+  /// Creates a render object for an iOS UIView.
+  ///
+  /// The `viewId`, `hitTestBehavior`, and `gestureRecognizers` parameters must not be null.
+  RenderUiKitView({required super.viewController, required super.hitTestBehavior, required Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers}) {
+    updateGestureRecognizers(gestureRecognizers);
+  }
+
+  /// {@macro flutter.rendering.PlatformViewRenderBox.updateGestureRecognizers}
+  void updateGestureRecognizers(Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers) {
```

Comment:
@cbracken and I discussed this a little as well. I believe our approach to gesture handling will need to be decoupled between iOS and macOS. Please feel free to clarify further if you remember.

Reply:
Would it make sense to wait with this refactoring then until we have an AppKit implementation and have clarity on what's common and what's not?

Reply:
As I understand, implementing AppKit would require either already having superclasses with common functionality in place, or duplicating everything that they would have in common, which was the impetus for this refactor.

Reply:
Gesture handling is not yet supported on macOS, but will likely be within the next week or two. Given that, perhaps we should push this up to the superclass now rather than later. API-wise this is pretty safe to support even before we actually have the embedder side work done I think. (We're cheating and testing by using UIKitView already as-is).

Reply:
I agree that it's better to push gesture stuff to the superclass. Or if you want to keep it here, add a TODO, and also a comment explaining why here.